src/enc/worker_pool.rs

```diff
@@ -132,7 +132,7 @@ impl <ReturnValue:Send+'static,
             if local_queue.shutdown{
               break;
             } else {
-              let _ = cvar.wait(local_queue); // unlock immediately, unfortunately
+              let _lock = cvar.wait(local_queue); // unlock immediately, unfortunately
```

Comment:
why do you need a named var?

Reply:
Because if the variable isn't named, we get this error:

```
error: non-binding let on a synchronization lock                                                                                                                                                                                                                                                    
   --> src\enc\worker_pool.rs:135:19
    |
135 |               let _ = cvar.wait(local_queue); // unlock immediately, unfortunately
    |                   ^ this lock is not assigned to a binding and is immediately dropped
```

Reply:
Thanks!  Does it conflict with the comment then? Since we don't unlock immediately?  I think the comment may need an update/explanation

Reply:
I think the comment still makes sense since the lock is dropped right after the continue on next line, isn't it?